### PR TITLE
Runtime query'able version information

### PR DESCRIPTION
**Pull request by DrusTheAxe**
Add new API to query Windows App SDK runtime version in use at runtime

`WindowsAppSDK-VersionInfo.h` at build-time provides what you're built with, but that's merely a "minimum criteria" what you'll use at runtime when using WinAppSDK framework-dependent (aka as MSIX packages). If you build with 1.2.3 you could run with 1.2.3, or 1.2.4, or 1.2.5, or... This lets you query for what is in use at runtime

See [Runtime query'able version information API spec #2893](https://github.com/microsoft/WindowsAppSDK/pull/2893) for the API spec

https://task.ms/37963921

### CODE REVIEW

File: dev/Common/WindowsAppRuntime.VersionInfo.cpp [hunk 1]
```diff
@@ -39,6 +46,22 @@ class RuntimeInformation
...
+    {
+        if (!g_versionInfo)
+        {
+            static wil::unique_hmodule module{ LoadResourceModule() };
+
+            auto getVersionInfo{ GetProcAddressByFunctionDeclaration(module.get(), WindowsAppRuntime_GetVersionInfo) };
+            THROW_LAST_ERROR_IF_NULL(getVersionInfo);
+
+            g_versionInfo = static_cast<const ::Microsoft::WindowsAppSDK::VersionInfo*>(getVersionInfo());
+
```
Comment (sachintaMSFT) [inline dev/Common/WindowsAppRuntime.VersionInfo.cpp:60]: NIT: Remove blank line#60.

**Review by sachintaMSFT — APPROVED** [review]
Approving with minor feedback.

File: test/DynamicDependency/data/Microsoft.WindowsAppRuntime.Framework/appxmanifest.xml [hunk 2]
```diff
@@ -54,12 +69,13 @@
       </InProcessServer>
     </Extension>
     <Extension Category="windows.activatableClass.inProcessServer">
-      <InProcessServer>
-        <Path>Microsoft.WindowsAppRuntime.dll</Path>
-        <ActivatableClass ActivatableClassId="Microsoft.Windows.ApplicationModel.WindowsAppRuntime.DeploymentManager" ThreadingModel="both" />
```
[the remark this answers is not in the document]
Comment (DrusTheAxe) [reply test/DynamicDependency/data/Microsoft.WindowsAppRuntime.Framework/appxmanifest.xml:59]: alphabetical for human comprehension / maintainability

File: build/NuSpecs/AppxManifest.xml [hunk 3]
```diff
@@ -49,22 +61,14 @@
...
+          <ActivatableClass ActivatableClassId="Microsoft.Windows.System.EnvironmentManager" ThreadingModel="both" />
 
           <!-- PowerNotifications -->
           <ActivatableClass ActivatableClassId="Microsoft.Windows.System.Power.PowerManager" ThreadingModel="both" />
-
-          <!-- AppNotifications -->
-          <ActivatableClass ActivatableClassId="Microsoft.Windows.AppNotifications.AppNotificationManager" ThreadingModel="both" />
-          <ActivatableClass ActivatableClassId="Microsoft.Windows.AppNotifications.AppNotification" ThreadingModel="both" />
-          <ActivatableClass ActivatableClassId="Microsoft.Windows.AppNotifications.AppNotificationProgressData" ThreadingModel="both" />
-          <ActivatableClass ActivatableClassId="Microsoft.Windows.AppNotifications.AppNotificationActivatedEventArgs" ThreadingModel="both" />
```
[the remark this answers is not in the document]
Comment (DrusTheAxe) [reply build/NuSpecs/AppxManifest.xml:64]: alphabetical for human comprehension / maintainability

File: test/Deployment/data/WindowsAppRuntime.Test.Framework/appxmanifest.xml [hunk 4]
```diff
@@ -47,18 +47,25 @@
...
-        <ActivatableClass ActivatableClassId="Microsoft.Windows.ApplicationModel.WindowsAppRuntime.DeploymentManager" ThreadingModel="both" />
-        <ActivatableClass ActivatableClassId="Microsoft.Windows.ApplicationModel.WindowsAppRuntime.DeploymentResult" ThreadingModel="both" />
-        <ActivatableClass ActivatableClassId="Microsoft.Windows.ApplicationModel.WindowsAppRuntime.DeploymentInitializeOptions" ThreadingModel="both" />
+        <ActivatableClass ActivatableClassId="Microsoft.Windows.ApplicationModel.WindowsAppRuntime.VersionInfo.ReleaseInfo" ThreadingModel="both" />
+        <ActivatableClass ActivatableClassId="Microsoft.Windows.ApplicationModel.WindowsAppRuntime.VersionInfo.RuntimeInfo" ThreadingModel="both" />
+      </InProcessServer>
+    </Extension>
+    <Extension Category="windows.activatableClass.inProcessServer">
+      <InProcessServer>
+        <Path>Microsoft.WindowsAppRuntime.dll</Path>
```
[the remark this answers is not in the document]
Comment (DrusTheAxe) [reply test/Deployment/data/WindowsAppRuntime.Test.Framework/appxmanifest.xml:64]: alphabetical for human comprehension / maintainability

File: test/DynamicDependency/data/Microsoft.WindowsAppRuntime.Framework/appxmanifest.xml [hunk 5]
```diff
@@ -54,12 +69,13 @@
...
-        <ActivatableClass ActivatableClassId="Microsoft.Windows.ApplicationModel.WindowsAppRuntime.DeploymentResult" ThreadingModel="both" />
-        <ActivatableClass ActivatableClassId="Microsoft.Windows.ApplicationModel.WindowsAppRuntime.DeploymentInitializeOptions" ThreadingModel="both" />
-      </InProcessServer>
+        <InProcessServer>
+            <Path>Microsoft.WindowsAppRuntime.dll</Path>
+            <ActivatableClass ActivatableClassId="Microsoft.Windows.AppNotifications.AppNotificationManager" ThreadingModel="both" />
+            <ActivatableClass ActivatableClassId="Microsoft.Windows.AppNotifications.AppNotification" ThreadingModel="both" />
```
[the remark this answers is not in the document]
Comment (DrusTheAxe) [reply test/DynamicDependency/data/Microsoft.WindowsAppRuntime.Framework/appxmanifest.xml:75]: I don't see that

Though GitHub's diff is awkward to see. View file shows no redundancy

image

**Comment by DrusTheAxe** [comment]
>Sorry! Our services aren't available right now.

UpdateUnreliableTests.ps1 lived up to its name. Kicking off another build to test the question, "Do you feel lucky?"...

**Pull request merged**